query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/QueryRequestToPromqlConverter.java

```diff
@@ -105,12 +104,10 @@ private QueryTimeRange getQueryTimeRange(ExecutionContext executionContext) {
             expression ->
                 expression.getValueCase().equals(ValueCase.FUNCTION)
                     && !QueryRequestUtil.isDateTimeFunction(expression))
-        .map(
-            functionExpression ->
-                ImmutablePair.of(
-                    PrometheusUtils.getColumnNameForMetricFunction(functionExpression),
-                    mapToPromqlQuery(functionExpression, groupByList, filterList, duration)))
-        .collect(Collectors.toMap(ImmutablePair::getLeft, ImmutablePair::getRight));
+        .collect(
```

Comment:
Ah! we missed this during the review. This was the case before we added this function `getColumnNameForMetricFunction` of fixing the same attribute with two different metric aggregation function situations - `select SERVICE.service_id, sum(SERVICE.numCalls), sum(SERVICE.errorCount)`

If ImmputablePair is not required anywhere, you can remove its lib deps from build too, can you check if it's used anywhere else?

Reply:
So `getColumnNameForMetricFunction` returns a unique value for `sum(SERVICE.numCalls)` and `sum(SERVICE.errorCount)`? If that's the case, it's not really the column name (which is `SERVICE.numCalls` in both cases), so maybe `generateAliasForMetricFunction` might be more appropriate. If it's not the case, this code would error in the above example.

Reply:
done